### ISSUE TITLE
Issue with Bun 1.0 (and a solution)

### ISSUE THREAD

**Issue by 7f8ddd** [mafintosh/stream-shift#4]
I recently migrated my project over to Bun to test it out, and encountered this error:
`TypeError: undefined is not an object (evaluating 'state.buffer[0].length')`

I solved this by changing this function:
```
function getStateLength (state) {
  if (state.buffer.length) {
    // Since node 6.3.0 state.buffer is a BufferList not an array
    if (state.buffer.head) {
      return state.buffer.head.data.length
    }
    
    return state.buffer[0].length
  }

  return state.length
}
```
Into this:
```
function getStateLength (state) {
  return state.length
}
```

Hope it helps someone!
This fix may be incorrect but it solved my issue with some library that was using it throwing an error on my website.

**Comment by cstrnt** [comment]
Same issue with Node 20.11 due to the following change (https://github.com/nodejs/node/commit/139d6c8d3b)

**Comment by mfulton26** [comment]
What about changing `return state.buffer[0].length` to account for `state.buffer`'s first element being `null` or `undefined`? That would address this issue for both Bun and Node 20.

e.g. `return state.buffer[0]?.length ?? state.buffer.length`

**Comment by henrik242** [comment]
@mafintosh Isn't this fixed in stream-shift 1.0.3?

**Comment by mafintosh** [comment]
should be ya
